File: Me/leetcode/链表环index/twoSum.js
var twoSum = function(nums, target) {
  const map = {}
  const len = nums.length
  for(let i=0;i<len; i++){
      const targetNum = target - nums[i]
      console.log(targetNum, targetNum in map, map)
      // in 索引的是 每一项的 索引号
      if(targetNum in map) return [map[targetNum], i]
      map[nums[i]] = i  
  }  
};

console.log(twoSum([1, 5 , 5, 8], 10))

File: Me/leetcode/python-leetcode/twoSum.py
class Solution:
     def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_ = {}
        for i in range(len(nums)):
            tmp = target - nums[i]
            if tmp in dict_:
                return [dict_[tmp], i]
            dict_[nums[i]]=i